operations/averageOperation.h

```diff
@@ -0,0 +1,29 @@
+/*
+    This file defines the AverageOperation class which represents the average of
+    multiple tensors.
+*/
+
+#include "operations/operation.h"
+
+#ifndef __OP_AVG_INCLUDED__   
+#define __OP_AVG_INCLUDED__   
+
+template <typename T>
+class AverageOperation : public Operation<T> {
+    public:
+
+    vector<Tensor<T*>> tensors;
```

Comment:
Thanks for the pull request!! This is a prelim review.

The representation of the Average Tensor will need to be changed I think. Generally how the network takes in a tensor is a single tensor with size [batch_size, input_dim]. SO the average operation should take as input t1 which would be a tensor and another int variable called axis specifying what dimension to average along.  

Hence the API would look like

```c++
loss = losses::mse(y,gt)
averageLoss = tensorOps::average(loss,axis=0)
// and then continue from here
```

This would be inline how pytorch, tf and numpy handle some operation along a list of objects. Hence, I think it would be better to adhere with their API.

Reply:
Right, I missed the big picture here. Thanks for pointing this out.

Reply:
No problem, looking forward to your updates :)